drivers/net/ppp.c

```diff
@@ -954,14 +954,14 @@ static void ppp_iface_init(struct net_if *iface)
 
 	memset(ppp->buf, 0, sizeof(ppp->buf));
 
-	/* If we have a GSM modem with PPP support or interface autostart is disabled
-	 * from Kconfig, then do not start the interface automatically but only
-	 * after the modem is ready or when manually started.
+	/* If the interface autostart is disabled from Kconfig, then do not
+	 * start the interface automatically but only after manually started.
 	 */
-	if (IS_ENABLED(CONFIG_MODEM_GSM_PPP) ||
-	    IS_ENABLED(CONFIG_PPP_NET_IF_NO_AUTO_START)) {
+	if (IS_ENABLED(CONFIG_PPP_NET_IF_NO_AUTO_START)) {
 		net_if_flag_set(iface, NET_IF_NO_AUTO_START);
 	}
+
+	net_if_carrier_off(iface);
```

Comment:
This might need some more testing, I think there was some reason for extra check but do not recall now what it was. Unfortunately I do not have any GSM modem that I could try atm, if any other reviewer has one, please test this with GSM modem.

Reply:
Unfortunately, I don't have a modem either to verify. But the situation here is similar to what Ethernet drivers did - they disabled interface autostart only to bring it up with `net_if_up()` once the carrier was detected. With those changes, it's no longer needed, as drivers can signalize the carrier state with `net_if_carrier_*()` API. The interface can be brought admin up during system boot, it's only needed that the carrier state is set to off during init, so that `NET_EVENT_IF_UP` is not signified prematurely.